asdf/asdftypes.py

```diff
@@ -25,6 +26,9 @@
                           list(six.integer_types) +
                           [float, list, dict, tuple])
 
+# regex used to parse module name from optional version string
+MODULE_RE = re.compile(r'([a-zA-Z]+)(-(\d.\d.\d))?')
```

Comment:
Yes, good catch. I was testing for a narrow case but I believe "\d+" for each of those is correct.